ddsrouter_core/src/cpp/participant/implementations/rtps/CommonParticipant.hpp

```diff
@@ -41,19 +41,19 @@ namespace rtps {
 /**
  * TODO
  */
-template <class ConfigurationType>
-class CommonRTPSRouterParticipant
-    : public BaseParticipant<ConfigurationType>
+class CommonParticipant
+    : public BaseParticipant
     , public fastrtps::rtps::RTPSParticipantListener
 {
 public:
 
-    CommonRTPSRouterParticipant(
-            const ConfigurationType participant_configuration,
+    CommonParticipant(
+            std::shared_ptr<configuration::SimpleParticipantConfiguration> participant_configuration,
```

Comment:
Maybe this would better be a `configuration::ParticipantConfiguration` ??

Reply:
Also related, what is the advantage of `DiscoveryServerConfiguration` being a child of `SimpleParticipantConfiguration`? Couldn't they be siblings?

Reply:
Changed so CommonParticipant get as arguments attributes and domain.

Reply:
About `DSConf` is child of `SimpleConf` because it reuses the domain. It could not, and have a fixed domain for all DS, but this does not help us anyway, and be able to set the domain could (lets try to avoid limiting us if not necessary)